File: scripts/tutorial7/python/plotAvgFailedTask.py
from plotGenericLine import plot_generic_line

if __name__ == '__main__':
    print("--- Generating: Average Failed Task Plots ---")
    
    # Group 1: Overall Failed Tasks
    plot_generic_line(row_offset=1, column_offset=2, y_label='Failed Tasks (%)',
                      app_type='ALL_APPS', calculate_percentage='percentage_of_all', legend_pos='upper left')
    plot_generic_line(row_offset=1, column_offset=2, y_label='Failed Tasks for\nAugmented Reality App (%)',
                      app_type='LLM_INFERENCE', calculate_percentage='percentage_of_all', legend_pos='upper left')


    # Group 2: Failed Tasks on Edge
    plot_generic_line(row_offset=2, column_offset=2, y_label='Failed Tasks on Edge (%)',
                      app_type='ALL_APPS', calculate_percentage='percentage_of_all', legend_pos='upper left')
    plot_generic_line(row_offset=2, column_offset=2, y_label='Failed Tasks on Edge for\nAugmented Reality App (%)',
                      app_type='LLM_INFERENCE', calculate_percentage='percentage_of_all', legend_pos='upper left')


    # Group 3: Failed Tasks on Cloud
    plot_generic_line(row_offset=3, column_offset=2, y_label='Failed Tasks on Cloud (%)',
                      app_type='ALL_APPS', calculate_percentage='percentage_of_all', legend_pos='upper left')
    plot_generic_line(row_offset=3, column_offset=2, y_label='Failed Tasks on Cloud for\nAugmented Reality App (%)',
                      app_type='LLM_INFERENCE', calculate_percentage='percentage_of_all', legend_pos='upper left')

File: scripts/tutorial7/matlab/plotAvgFailedTask.m
function [] = plotAvgFailedTask()

    plotGenericLine(1, 2, 'Failed Tasks (%)', 'ALL_APPS', 'percentage_of_all', 'NorthWest');
    plotGenericLine(1, 2, {'Failed Tasks for';'Disaster Map Fusion App (%)'}, 'DISASTER_MAP_FUSION', 'percentage_of_all', 'NorthWest');
    plotGenericLine(1, 2, 'Failed Tasks for Situation Awareness Alerts App (%)', 'SITUATION_AWARENESS_ALERTS', 'percentage_of_all', 'NorthWest');

    plotGenericLine(2, 2, 'Failed Tasks on Edge (%)', 'ALL_APPS', 'percentage_of_all', 'NorthWest');
    plotGenericLine(2, 2, {'Failed Tasks on Edge';'for Disaster Map Fusion App (%)'}, 'DISASTER_MAP_FUSION', 'percentage_of_all', 'NorthWest');
    plotGenericLine(2, 2, 'Failed Tasks on Edge for Situation Awareness Alerts App (%)', 'SITUATION_AWARENESS_ALERTS', 'percentage_of_all', 'NorthWest');

    plotGenericLine(3, 2, 'Failed Tasks on Cloud (%)', 'ALL_APPS', 'percentage_of_all', 'NorthWest');
    plotGenericLine(3, 2, {'Failed Tasks on Cloud for';'Disaster Map Fusion App (%)'}, 'DISASTER_MAP_FUSION', 'percentage_of_all', 'NorthWest');
    plotGenericLine(3, 2, 'Failed Tasks on Cloud for Situation Awareness Alerts App (%)', 'SITUATION_AWARENESS_ALERTS', 'percentage_of_all', 'NorthWest');
    
end